### PR TITLE
Publish NuGet Packages for Canary Builds to NuGet.org

### PR DESCRIPTION
**Pull request by NickGerleman**
Publishing our latest WinMDs would allow the docs-site to do automated document generation on its side. Do not limited code signing tasks to stable branches.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/react-native-windows/pull/9096)

### CODE REVIEW

File: .ado/templates/prep-and-pack-single.yml [hunk 1]
```diff
@@ -30,6 +30,7 @@ steps:
 
   - ${{ if eq(parameters.codesignBinaries, true) }}:
     - task: SFP.build-tasks.custom-build-task-1.EsrpCodeSigning@1
+      condition: eq($(Build.Repository.ID), 'microsoft/react-native-windows')
```
Comment (NickGerleman) [inline .ado/templates/prep-and-pack-single.yml:33]: Verified this variable is correct on publish on microsoft instance.

**Comment by asklar** [comment]
We currently publish canary nugets to the ADO public feed:
https://dev.azure.com/ms/react-native/_packaging?_a=package&feed=react-native-public&package=Microsoft.ReactNative&protocolType=NuGet&version=0.0.0-canary.410&view=overview

However those bits only contain Release binaries, which do not carry the documentation strings in them.

Publish was building Debug as part of its pipeline but to make publishing faster this was removed.
It seems like we may need to revisit that if we want to publish Debug nuget packages for generating the markdown docs?

**Review by asklar — CHANGES_REQUESTED** [review]
We currently publish canary nugets to the ADO public feed:
https://dev.azure.com/ms/react-native/_packaging?_a=package&feed=react-native-public&package=Microsoft.ReactNative&protocolType=NuGet&version=0.0.0-canary.410&view=overview

However those bits only contain Release binaries, which do not carry the documentation strings in them.

Publish was building Debug as part of its pipeline but to make publishing faster this was removed.

It seems like we may need to revisit that if we want to publish Debug nuget packages for generating the markdown docs?

**Comment by NickGerleman** [comment]
@asklar  debug packages are separately added with https://github.com/microsoft/react-native-windows/pull/9097

**Comment by NickGerleman** [comment]
@asklar @dannyvv @jonthysell could you take a look at this again, understanding that the scope of this is to publish canary packages to NuGet.org?

**Comment by jonthysell** [comment]
> @asklar Alexander Sklar FTE @dannyvv Danny van Velzen FTE @jonthysell Jon Thysell (JAUNTY) FTE could you take a look at this again, understanding that the scope of this is to publish canary packages to NuGet.org?

Why do we need to publish the canary packages to NuGet.org? What's wrong with just publishing them to our public ADO feed and letting people get them from there if they need them?

**Comment by asklar** [comment]
in reality if all we care about is the winmd, we don't need a nuget package, we could just upload the winmd to an azure artifact too. And we could additionally not actually build all of microsoft.reactnative.dll, and just run MIDL, which would bring down the time this takes to hopefully only a minute or two.

**Comment by jonthysell** [comment]
> in reality if all we care about is the winmd, we don't need a nuget package, we could just upload the winmd to an azure artifact too. And we could additionally not actually build all of microsoft.reactnative.dll, and just run MIDL, which would bring down the time this takes to hopefully only a minute or two.

Having debug packages still gets us better support for troubleshooting customers.

Where do we want the winmd to be processed? Is it here in the repo or in the samples repo? The GH pipelines in the samples repo may find it easier to grab it from the nuget than downloading from ADO.

**Comment by asklar** [comment]
> > in reality if all we care about is the winmd, we don't need a nuget package, we could just upload the winmd to an azure artifact too. And we could additionally not actually build all of microsoft.reactnative.dll, and just run MIDL, which would bring down the time this takes to hopefully only a minute or two.
> 
> Having debug packages still gets us better support for troubleshooting customers.
> 
> Where do we want the winmd to be processed? Is it here in the repo or in the samples repo? The GH pipelines in the samples repo may find it easier to grab it from the nuget than downloading from ADO.

we should process it in the samples repo.
I don't expect we'd benefit from having a debug nuget package: customers using nuget would use the release RNW bits, or build from source. What scenario would a customer hit that would have them use the RNW debug nuget (and how?)

**Comment by NickGerleman** [comment]
Time isn't an issue, and the PR to add debug nuget builds is merged. Debug builds were originally removed from publish because its artifacts were intermediate, meant for publish, and then they weren't published.

> Why do we need to publish the canary packages to NuGet.org? What's wrong with just publishing them to our public ADO feed and letting people get them from there if they need them?

In a similar vain, why are any of our packages on NuGet.org? It means not having to deal with multiple feeds, or that sort of headache.

**Comment by asklar** [comment]
> Time isn't an issue, and the PR to add debug nuget builds is merged. Debug builds were originally removed from publish because its artifacts were intermediate, meant for publish, and then they weren't published.
> 
> > Why do we need to publish the canary packages to NuGet.org? What's wrong with just publishing them to our public ADO feed and letting people get them from there if they need them?
> 
> In a similar vain, why are any of our packages on NuGet.org? It means not having to deal with multiple feeds, or that sort of headache.

who is going to use the debug nugets, and how, though? today you get the nugets in your app if you pass the option in the CLI. How does a customer get to use the debug nugets (and more importantly, why). I think there is no customer scenario for debug nuget, but it may be that we use debug nuget as an easy way to upload a binary "somewhere". However I agree that this doesn't need to go into nuget.org, doing so can create confusion/pollute the set of versions available (e.g. "do I need to use the debug nuget if I want to debug my app?")

**Comment by NickGerleman** [comment]
Debug packages are already set to go to NuGet.org. See above for the scenario Jon mentioned about why the debug package exists. This PR only concerns canary packages, and whether they make it there.

**Comment by dannyvv** [comment]
Note I recall seeing some mention on aka.ms/nuget about not pushing rolling or daily builds and that that should go, but I can't find the reference, so maybe that guidance has been retracted or I may be misinformed.

**Comment by NickGerleman** [comment]
Sounds like there are some strong opinions on whether this should go on NuGet.org. Going to hold off on this PR until a greater need comes up.

**Pull request closed without merging**